index.js

```diff
@@ -56,8 +56,12 @@ module.exports = function (request, options) {
 
     function onEnd(err) {
       if(err) reject(err);
-      cleanup();
-      return resolve({fields, files});
+      Promise.all(filePromises)
```

Comment:
@dominhhai, good point, but I am wondering if that is worth adding branching in the code.  Let's put a pin on that, if perf ever becomes a problem, we can pull this back. Thanks for the insight!